### ISSUE TITLE
Accessibility Issue - NVDA does not announce selected option in mobile or tablet view.

### ISSUE THREAD

**Issue by anoopmundathan** [JedWatson/react-select#5790]
react-select - Version 5.7.7
Browser - Microsoft Edge for business 118.0.2088.69
Screen reader - NVDA - Version 2023 (2023.1.0.27913)

I have a simple select component as below
```
import Select from 'react-select'

const options = [
  { value: '', label: 'blank' },
  { value: 'spanish', label: 'Spanish' },
  { value: 'french', label: 'French' },
  { value: 'italian', label: 'Italian' },
  { value: 'german', label: 'German' }
];

export const App = () => { 
  return (
    <div className="App">
      <Select options={options} />
    </div>
  );

```
When I tested this component against NVDA screen reader, for desktop view NVDA correctly announces the selection but not for iPad or mobile view it does not, it announces title of the page (here React App document). Please see screenshot

**Desktop**

<img width="697" alt="Screenshot 2023-10-27 at 11 04 54" src="https://github.com/JedWatson/react-select/assets/3778229/7c396cf1-6af3-46cf-9feb-195b4708b9f9">

**iPad view**

<img width="669" alt="Screenshot 2023-10-27 at 11 05 23" src="https://github.com/JedWatson/react-select/assets/3778229/1a768d31-b4df-43ed-967b-ec13294e88f1">

Please include a test case that demonstrates the issue you're reporting!

This is very helpful to maintainers in order to help us see the issue you're seeing.

Please note we are currently only directing our efforts towards the current major (v5) version and beyond.

We understand this might be inconvenient but it is in the best interest of supporting the broader community and to sustain the `react-select` project going forward.

To report bugs against react-select v5 please fork the following code-sandbox:
https://codesandbox.io/s/react-select-v5-sandbox-y5jtm

You may also find the [online Babel tool](https://babeljs.io/repl/) quite helpful if you wish to use ES6/ES7 syntax not yet supported by the browser you are using.
